To: vim_dev@googlegroups.com
Subject: Patch 7.3.117
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.117
Problem:    On some systems --as-needed does not work, because the "tinfo"
	    library is included indirectly from "ncurses". (Charles Campbell)
Solution:   In configure prefer using "tinfo" instead of "ncurses".
Files:	    src/configure.in, src/auto/configure


*** ../vim-7.3.116/src/configure.in	2010-11-16 19:25:56.000000000 +0100
--- src/configure.in	2011-02-09 17:28:16.000000000 +0100
***************
*** 2618,2630 ****
    AC_MSG_RESULT([empty: automatic terminal library selection])
    dnl  On HP-UX 10.10 termcap or termlib should be used instead of
    dnl  curses, because curses is much slower.
!   dnl  Newer versions of ncurses are preferred over anything.
    dnl  Older versions of ncurses have bugs, get a new one!
    dnl  Digital Unix (OSF1) should use curses (Ronald Schild).
    dnl  On SCO Openserver should prefer termlib (Roger Cornelius).
    case "`uname -s 2>/dev/null`" in
! 	OSF1|SCO_SV)	tlibs="ncurses curses termlib termcap";;
! 	*)	tlibs="ncurses termlib termcap curses";;
    esac
    for libname in $tlibs; do
      AC_CHECK_LIB(${libname}, tgetent,,)
--- 2618,2631 ----
    AC_MSG_RESULT([empty: automatic terminal library selection])
    dnl  On HP-UX 10.10 termcap or termlib should be used instead of
    dnl  curses, because curses is much slower.
!   dnl  Newer versions of ncurses are preferred over anything, except
!   dnl  when tinfo has been split off, it conains all we need.
    dnl  Older versions of ncurses have bugs, get a new one!
    dnl  Digital Unix (OSF1) should use curses (Ronald Schild).
    dnl  On SCO Openserver should prefer termlib (Roger Cornelius).
    case "`uname -s 2>/dev/null`" in
! 	OSF1|SCO_SV)	tlibs="tinfo ncurses curses termlib termcap";;
! 	*)	tlibs="tinfo ncurses termlib termcap curses";;
    esac
    for libname in $tlibs; do
      AC_CHECK_LIB(${libname}, tgetent,,)
*** ../vim-7.3.116/src/auto/configure	2010-11-16 19:25:56.000000000 +0100
--- src/auto/configure	2011-02-09 17:29:13.000000000 +0100
***************
*** 9886,9894 ****
  else
    { $as_echo "$as_me:${as_lineno-$LINENO}: result: empty: automatic terminal library selection" >&5
  $as_echo "empty: automatic terminal library selection" >&6; }
!               case "`uname -s 2>/dev/null`" in
! 	OSF1|SCO_SV)	tlibs="ncurses curses termlib termcap";;
! 	*)	tlibs="ncurses termlib termcap curses";;
    esac
    for libname in $tlibs; do
      as_ac_Lib=`$as_echo "ac_cv_lib_${libname}''_tgetent" | $as_tr_sh`
--- 9886,9894 ----
  else
    { $as_echo "$as_me:${as_lineno-$LINENO}: result: empty: automatic terminal library selection" >&5
  $as_echo "empty: automatic terminal library selection" >&6; }
!                 case "`uname -s 2>/dev/null`" in
! 	OSF1|SCO_SV)	tlibs="tinfo ncurses curses termlib termcap";;
! 	*)	tlibs="tinfo ncurses termlib termcap curses";;
    esac
    for libname in $tlibs; do
      as_ac_Lib=`$as_echo "ac_cv_lib_${libname}''_tgetent" | $as_tr_sh`
*** ../vim-7.3.116/src/version.c	2011-02-09 17:09:26.000000000 +0100
--- src/version.c	2011-02-09 17:41:37.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     117,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
220. Your wife asks for sex and you tell her where to find you on IRC.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
